src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.tsx

```diff
@@ -23,12 +27,25 @@ function BaseSidebarScreen() {
     const styles = useThemeStyles();
     const activeWorkspaceID = useActiveWorkspaceFromNavigationState();
     const {translate} = useLocalize();
+    const didResetActiveWorkspaceIDRef = useRef(false);
 
     useEffect(() => {
         Performance.markStart(CONST.TIMING.SIDEBAR_LOADED);
         Timing.start(CONST.TIMING.SIDEBAR_LOADED, true);
     }, []);
 
+    const [selectedPolicy] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${activeWorkspaceID ?? -1}`);
+
+    useEffect(() => {
+        if (!!selectedPolicy || activeWorkspaceID === undefined || !!didResetActiveWorkspaceIDRef.current) {
+            return;
+        }
+
+        Navigation.navigateWithSwitchPolicyID({policyID: undefined});
+        updateLastAccessedWorkspace(undefined);
+        didResetActiveWorkspaceIDRef.current = true;
```

Comment:
Why do we need to set this ref? What causes the effect to run again?

Reply:
Can you please check this in depth and what values are being logged on the dependences when the effect run again

Reply:
Here are the dependence values. As you can see, there are two times the ```activeWorkspace``` is ```undefined``` and ```activeWorkspaceID``` is ```xxxxxxxxxxx```.

![image](https://github.com/user-attachments/assets/7191264f-016d-4e3a-94bb-5848b687efa5)

Reply:
That's unexpected. Why is useEffect called twice if the dependencies didn't change?

Reply:
I think it is because of the strict mode. Testing to verify it

Reply:
I turned off strict mode and it works well:

![image](https://github.com/user-attachments/assets/fbfcee82-b7c0-4bb0-803d-7b9fd2242293)

Reply:
Then can we remove the ref?

Reply:
I removed ```didResetActiveWorkspaceIDRef```